src/audio.c

```diff
@@ -0,0 +1,212 @@
+/*
```

Comment:
Yet again can this go into its own `system/`-like folder with different implementations? Also with the encoder split out to its own file called something like `rsAACEncoder`

Reply:
you want to move the whole of audio.c to system ? I'd say SDL2 is already a system abstraction. If it's only about moving the file to the directory fine by me, but I don't think this requires an abstraction like you did for X11

Reply:
Not specifically to `system` but a new folder that is similar. Despite already using SDL2 which is a system abstraction I might in the future look at other libraries and implementations.

Reply:
so what directory name do you propose ?

Reply:
`audio_input` and the encoder in a file called `audio_encoder`

Reply:
tried to do that but came into a problem... AudioEncoder has to include Audio, since the functions require RSAudio pointer. Audio has to include AudioEncoder since it creates it. Unlike in c++ I can't do a forward declaration of Audio for AudioEncoder, at least not as long as the struct is typedef'd or without a warring (redefition of typedef RSAudio)

Reply:
Why does rsaudio have to create the encoder? Why does the encoder need to have rsaudio? Like I've said just have an extract method (which would be part of the audio circle) to pass data from one thing to another.

Reply:
I have an idea how to separate it I'll do it tomorrow (well technically today it's past midnight )